### PR TITLE
add e2e test for InterpretHealth

### PR DESCRIPTION
**Pull request by zhuwint**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Part of #2354 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
NONE

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
Thanks~
/assign

File: test/e2e/resourceinterpreter_test.go [hunk 1]
```diff
@@ -226,4 +227,46 @@ var _ = ginkgo.Describe("Resource interpreter webhook testing", func() {
...
+					memberWorkload.Status.ReadyReplicas = readyReplicas
+					framework.UpdateWorkload(clusterDynamicClient, memberWorkload, cluster, "status")
+				}
+			}
+
+			CheckResult := func(result workv1alpha2.ResourceHealth) interface{} {
+				return func(g gomega.Gomega) (bool, error) {
+					rb, err := karmadaClient.WorkV1alpha2().ResourceBindings(workload.Namespace).Get(context.TODO(), resourceBindingName, metav1.GetOptions{})
+					g.Expect(err).NotTo(gomega.HaveOccurred())
+					g.Expect(len(rb.Status.AggregatedStatus)).Should(gomega.Equal(3))
```
Comment (XiShanYongYe-Chang) [inline test/e2e/resourceinterpreter_test.go:249]: If this assertion fails, the loop wait will exit, which should not be what we want.
Besides, we can use `len(framework.ClusterNames())` to replace number 3.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/lgtm
/approve

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/lgtm

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**